package/MDAnalysis/analysis/waterdynamics.py

```diff
@@ -0,0 +1,1057 @@
+            #parallel selection to be implemented
+            selection_out = self._selection_serial(self.universe,self.selection)
+        self.timeseries = []
+        for dt in list(range(1,self.dtmax+1)):
+            output = self._getMeanOneCOHPoint(self.universe,selection_out,self.selection,dt,self.tf)
+            self.timeseries.append(output)
+
+    
+       
+class  AD:
+    r"""
+    The angular distribution function (AD) is defined as the distribution
+    probability of the cosine of the :math:`\theta` angle formed by the OH vector, HH vector
+    or dipolar vector of water molecules and a vector :math:`\hat n` parallel to chosen axis 
+    (z is the default value). The cosine is define as :math:`\cos \theta = \hat u \cdot \hat n`, where :math:`\hat u` is OH, HH or dipole vector. 
+    It creates a histogram and returns a list of lists, see Output_. The AD is also know as Angular Probability (AP).
+        
+    :Arguments:
+         *universe*
+             Universe object
+         *selection* 
+             Selection string to evaluate its angular distribution [‘byres name OH2’]
+         *bins*
+             Number of bins to create the histogram by means of numpy.histogram_ [40]
+         *axis*
+             Axis to create angle with the vector (HH, OH or dipole) and calculate cosine theta ['z']. Options: 'x',
+             'y', 'z'
+    
+    .. _numpy.histogram: http://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html
+
+
+    """
+    def __init__(self,universe,selection_str,bins=40,nproc=1,axis="z"):
+        self.universe = universe
+        self.selection_str = selection_str
+        self.bins = bins
+        self.nproc = nproc
+        self.axis = axis
+        self.graph = None
+
+    def _getCosTheta(self,universe,selection,axis):
+        valOH = []
+        valHH = []
+        valdip= []
+        i = 0        
+        while i <= (len(selection)-1):        
+            universe.trajectory[i]
+            line = selection[ i ].positions
+            j=0
+            while j < len(line):
+                Ot0 = numpy.array(line[j])
+                H1t0 = numpy.array(line[j+1])
+                H2t0 = numpy.array(line[j+2])
+
+                OHVector0 = H1t0 - Ot0
+                HHVector0 = H1t0 - H2t0
+                dipVector0 = (H1t0 + H2t0)*0.5 - Ot0
+                        
+                normOHVector0 = numpy.linalg.norm(OHVector0)
+                normHHVector0 = numpy.linalg.norm(HHVector0)
+                normdipVector0 = numpy.linalg.norm(dipVector0)
+                
+                unitOHVector0 = [OHVector0[0]/normOHVector0,OHVector0[1]/normOHVector0,OHVector0[2]/normOHVector0]
+                unitHHVector0 = [HHVector0[0]/normHHVector0,HHVector0[1]/normHHVector0,HHVector0[2]/normHHVector0]
+                unitdipVector0 = [dipVector0[0]/normdipVector0,dipVector0[1]/normdipVector0,dipVector0[2]/normdipVector0]
+                
+                if axis == "z":
```

Comment:
If you're doing a dot product with a unit vector, you can just slice that vector out, so 

np.dot(vector, [0, 0, 1]) === vector[2]